### PR TITLE
feat: forward port of #19153, using the `max u v =?= max u ?v` universe unification hack

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by hrmacbeth** [comment]
https://leanprover-community.github.io/mathlib-port-status/file/category_theory/limits/shapes/types is supposed to be out of sync from https://github.com/leanprover-community/mathlib/pull/19153 but I don't see any updates to that file from you on this PR.  Is it an oversight in the forward-port?

(First time reviewing a nontrivial forward-port, sorry if I'm misunderstanding something.)

**Review by Parcly-Taxel — APPROVED** [review]

File: Mathlib/Topology/Sheaves/PresheafOfFunctions.lean [hunk 1]
```diff
@@ -57,14 +58,14 @@ set_option linter.uppercaseLean3 false in
 #align Top.presheaf_to_Types TopCat.presheafToTypes
 
 @[simp]
-theorem presheafToTypes_obj {T : X → Type v} {U : (Opens X)ᵒᵖ} :
+theorem presheafToTypes_obj {T : X → Type w} {U : (Opens X)ᵒᵖ} :
```
Comment (hrmacbeth) [inline Mathlib/Topology/Sheaves/PresheafOfFunctions.lean:61]: This change doesn't occur in the mathlib3 PR.

File: Mathlib/Topology/Sheaves/PresheafOfFunctions.lean [hunk 2]
```diff
@@ -57,14 +58,14 @@ set_option linter.uppercaseLean3 false in
...
     (presheafToTypes X T).obj U = ∀ x : U.unop, T x :=
   rfl
 set_option linter.uppercaseLean3 false in
 #align Top.presheaf_to_Types_obj TopCat.presheafToTypes_obj
 
 @[simp]
-theorem presheafToTypes_map {T : X → Type v} {U V : (Opens X)ᵒᵖ} {i : U ⟶ V} {f} :
+theorem presheafToTypes_map {T : X → Type w} {U V : (Opens X)ᵒᵖ} {i : U ⟶ V} {f} :
```
Comment (hrmacbeth) [inline Mathlib/Topology/Sheaves/PresheafOfFunctions.lean:68]: This change doesn't occur in the mathlib3 PR.

File: Mathlib/Topology/Sheaves/PresheafOfFunctions.lean [hunk 3]
```diff
@@ -90,21 +91,23 @@ set_option linter.uppercaseLean3 false in
 #align Top.presheaf_to_Type TopCat.presheafToType
 
 @[simp]
-theorem presheafToType_obj {T : Type v} {U : (Opens X)ᵒᵖ} :
+theorem presheafToType_obj {T : Type w} {U : (Opens X)ᵒᵖ} :
```
Comment (hrmacbeth) [inline Mathlib/Topology/Sheaves/PresheafOfFunctions.lean:94]: This change doesn't occur in the mathlib3 PR.

File: Mathlib/Topology/Sheaves/PresheafOfFunctions.lean [hunk 4]
```diff
@@ -90,21 +91,23 @@ set_option linter.uppercaseLean3 false in
...
     (presheafToType X T).obj U = (U.unop → T) :=
   rfl
 set_option linter.uppercaseLean3 false in
 #align Top.presheaf_to_Type_obj TopCat.presheafToType_obj
 
 @[simp]
-theorem presheafToType_map {T : Type v} {U V : (Opens X)ᵒᵖ} {i : U ⟶ V} {f} :
+theorem presheafToType_map {T : Type w} {U V : (Opens X)ᵒᵖ} {i : U ⟶ V} {f} :
```
Comment (hrmacbeth) [inline Mathlib/Topology/Sheaves/PresheafOfFunctions.lean:101]: This change doesn't occur in the mathlib3 PR.

File: Mathlib/Topology/Sheaves/SheafCondition/UniqueGluing.lean [hunk 5]
```diff
@@ -84,23 +84,27 @@ We prove this to be equivalent to the usual one below in
 `TopCat.Presheaf.isSheaf_iff_isSheafUniqueGluing`
 -/
 def IsSheafUniqueGluing : Prop :=
-  ∀ ⦃ι : Type v⦄ (U : ι → Opens X) (sf : ∀ i : ι, F.obj (op (U i))),
-    IsCompatible F U sf → ∃! s : F.obj (op (iSup U)), IsGluing F U sf s
+  ∀ ⦃ι : Type w⦄ (U : ι → Opens X) (sf : ∀ i : ι, F.obj (op (U i))),
+    IsCompatible.{u, v, w} F U sf → ∃! s : F.obj (op (iSup U)), IsGluing.{u, v, w} F U sf s
```
Comment (hrmacbeth) [inline Mathlib/Topology/Sheaves/SheafCondition/UniqueGluing.lean:88]: You didn't specify the `.{u, v, w}` in the mathlib3 PR.  I'll trust that you did it right.

**Review by hrmacbeth — COMMENTED** [review]
Apart from the points mentioned, and a couple of places where you made porting notes explaining that a change was deliberate, I believe the statement changes have been ported correctly.  I didn't look closely at changes in proofs.

(Also, you didn't update the SHAs, but probably you know that.)

**Comment by semorrison** [comment]
Thanks @hrmacbeth. The three new changes you noted are correct, and should have been done back in https://github.com/leanprover-community/mathlib/pull/19153. We could backport them?

There are no changes to https://leanprover-community.github.io/mathlib-port-status/file/category_theory/limits/shapes/types  here as the changes corresponding to https://github.com/leanprover-community/mathlib/pull/19153 were made earlier in #5535.

I'm seriously considering just reverting https://github.com/leanprover-community/mathlib/pull/19153 at this point, it is a huge hassle, and still unclear that we have a viable way forward.

**Pull request closed without merging**